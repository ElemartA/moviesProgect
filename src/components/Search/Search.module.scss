.form {
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  .input {
    height: 40px;
    border-radius: 50px;
  }
  .fa{
    width: 80px;
    height: 40px;
    border-radius: 50px;
  }
}