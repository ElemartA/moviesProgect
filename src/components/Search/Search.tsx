import React, { FormEventHandler } from "react";
import type { FunctionComponent } from "react";
import s from "./Search.module.scss";
import { Button } from "react-bootstrap";

type TSearch = {
  value: string;
  onChange: (value: string) => void;
  onSubmit: FormEventHandler<HTMLFormElement> | undefined;
};

const Search: FunctionComponent<TSearch> = ({ value, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit} className={s.form} action="">
      <input
        style={{ width: 250 }}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        className="form-control"
        placeholder="введите запрос"
        aria-label="Поиск"
        type="search"
        data-testid="search"
      ></input>
      <Button type="submit" variant="secondary">
        Искать
      </Button>{" "}
    </form>
  </div>
);

export default Search;
