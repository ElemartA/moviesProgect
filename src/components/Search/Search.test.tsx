import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Search from "./Search";

describe("Search", () => {
  it("renders Search component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Search
          value={""}
          onChange={function (): void {
            throw new Error("Function not implemented.");
          }}
          onSubmit={undefined}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
    expect(screen.getByRole("searchbox")).toBeInTheDocument();
    expect(screen.queryByText(/Главная/i)).toBeNull();
  });
});

test("INPUT EVENT", () => {
  render(
    <Search
      value={""}
      onChange={function (): void {
        throw new Error("Function not implemented.");
      }}
      onSubmit={undefined}
    />
  );
  const input = screen.getByTestId("search") as HTMLInputElement;
  expect(input).toContainHTML("");
  fireEvent.blur(input, {
    target: { value: "тестовая строка" },
  });
  expect(input).toHaveValue("тестовая строка");
});
