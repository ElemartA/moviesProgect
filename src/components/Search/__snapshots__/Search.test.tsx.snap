// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component 1`] = `
<DocumentFragment>
  <div>
    <form
      action=""
      class="form"
    >
      <input
        aria-label="Поиск"
        class="form-control"
        data-testid="search"
        placeholder="введите запрос"
        style="width: 250px;"
        type="search"
        value=""
      />
      <button
        class="btn btn-secondary"
        type="submit"
      >
        Искать
      </button>
       
    </form>
  </div>
</DocumentFragment>
`;
